src/vocab.rs

```diff
@@ -34,6 +35,39 @@ impl PyVocab {
             }
         })
     }
+
+    fn ngram_indices(&self, word: &str) -> PyResult<Vec<Py<PyTuple>>> {
```

Comment:
`(String, usize)` implements direct conversion into `PyTuple`, so the return type should be `PyResult<Option<Vec<(String, usize)>>>`. 
I would also prefer returning an `Option<Vec<_>>` because explicit NGram vocabs could be unable to produce `(ngram, index)` tuples.